diff -uNr A/linux-3.15-rc4/arch/x86/boot/compressed/Makefile linux-3.15-rc4/arch/x86/boot/compressed/Makefile
--- A/linux-3.15-rc4/arch/x86/boot/compressed/Makefile	2014-05-04 21:14:42.000000000 -0400
+++ linux-3.15-rc4/arch/x86/boot/compressed/Makefile	2014-05-05 18:31:16.219088420 -0400
@@ -7,7 +7,7 @@
 targets := vmlinux vmlinux.bin vmlinux.bin.gz vmlinux.bin.bz2 vmlinux.bin.lzma \
 	vmlinux.bin.xz vmlinux.bin.lzo vmlinux.bin.lz4
 
-KBUILD_CFLAGS := -m$(BITS) -D__KERNEL__ $(LINUX_INCLUDE) -O2
+KBUILD_CFLAGS := -m$(BITS) -D__KERNEL__ $(LINUX_INCLUDE) -O0
 KBUILD_CFLAGS += -fno-strict-aliasing -fPIC
 KBUILD_CFLAGS += -DDISABLE_BRANCH_PROFILING
 cflags-$(CONFIG_X86_32) := -march=i386
@@ -36,6 +36,8 @@
 	VMLINUX_OBJS += $(obj)/eboot.o $(obj)/efi_stub_$(BITS).o
 endif
 
+VMLINUX_OBJS += /opt/intel/composer_xe_2013_sp1.2.144/compiler/lib/ia32/libirc_s.a
+
 $(obj)/vmlinux: $(VMLINUX_OBJS) FORCE
 	$(call if_changed,ld)
 	@:
diff -uNr A/linux-3.15-rc4/Makefile linux-3.15-rc4/Makefile
--- A/linux-3.15-rc4/Makefile	2014-05-04 21:14:42.000000000 -0400
+++ linux-3.15-rc4/Makefile	2014-05-05 18:31:16.239089457 -0400
@@ -243,9 +243,9 @@
 	  else if [ -x /bin/bash ]; then echo /bin/bash; \
 	  else echo sh; fi ; fi)
 
-HOSTCC       = gcc
-HOSTCXX      = g++
-HOSTCFLAGS   = -Wall -Wmissing-prototypes -Wstrict-prototypes -O2 -fomit-frame-pointer
+HOSTCC       = icc
+HOSTCXX      = icpc
+HOSTCFLAGS   = -Wall -Wmissing-prototypes -Wstrict-prototypes -O2 -no-vec -fomit-frame-pointer
 HOSTCXXFLAGS = -O2
 
 ifeq ($(shell $(HOSTCC) -v 2>&1 | grep -c "clang version"), 1)
@@ -349,7 +349,7 @@
 
 AS		= $(CROSS_COMPILE)as
 LD		= $(CROSS_COMPILE)ld
-CC		= $(CROSS_COMPILE)gcc
+CC		= icc
 CPP		= $(CC) -E
 AR		= $(CROSS_COMPILE)ar
 NM		= $(CROSS_COMPILE)nm
@@ -705,9 +705,6 @@
 # warn about C99 declaration after statement
 KBUILD_CFLAGS += $(call cc-option,-Wdeclaration-after-statement,)
 
-# disable pointer signed / unsigned warnings in gcc 4.0
-KBUILD_CFLAGS += $(call cc-disable-warning, pointer-sign)
-
 # disable invalid "can't wrap" optimizations for signed / pointers
 KBUILD_CFLAGS	+= $(call cc-option,-fno-strict-overflow)
 
diff -uNr A/linux-3.15-rc4/scripts/link-vmlinux.sh linux-3.15-rc4/scripts/link-vmlinux.sh
--- A/linux-3.15-rc4/scripts/link-vmlinux.sh	2014-05-04 21:14:42.000000000 -0400
+++ linux-3.15-rc4/scripts/link-vmlinux.sh	2014-05-05 18:31:16.245756469 -0400
@@ -42,7 +42,7 @@
 modpost_link()
 {
 	${LD} ${LDFLAGS} -r -o ${1} ${KBUILD_VMLINUX_INIT}                   \
-		--start-group ${KBUILD_VMLINUX_MAIN} --end-group
+		--start-group ${KBUILD_VMLINUX_MAIN} "/opt/intel/composer_xe_2013_sp1.2.144/compiler/lib/ia32/libirc_s.a" --end-group
 }
 
 # Link of vmlinux
@@ -50,14 +50,15 @@
 # ${2} - output file
 vmlinux_link()
 {
+	KBUILD_VMLINUX_MAIN="$KBUILD_VMLINUX_MAIN /opt/intel/composer_xe_2013_sp1.2.144/compiler/lib/ia32/libirc_s.a"
 	local lds="${objtree}/${KBUILD_LDS}"
 
 	if [ "${SRCARCH}" != "um" ]; then
-		${LD} ${LDFLAGS} ${LDFLAGS_vmlinux} -o ${2}                  \
+		${LD} ${LDFLAGS} ${LDFLAGS_vmlinux} -x -o ${2}                  \
 			-T ${lds} ${KBUILD_VMLINUX_INIT}                     \
 			--start-group ${KBUILD_VMLINUX_MAIN} --end-group ${1}
 	else
-		${CC} ${CFLAGS_vmlinux} -o ${2}                              \
+		${CC} ${CFLAGS_vmlinux} -s -x -o ${2}                              \
 			-Wl,-T,${lds} ${KBUILD_VMLINUX_INIT}                 \
 			-Wl,--start-group                                    \
 				 ${KBUILD_VMLINUX_MAIN}                      \
diff -uNr A/linux-3.15-rc4/scripts/Makefile.build linux-3.15-rc4/scripts/Makefile.build
--- A/linux-3.15-rc4/scripts/Makefile.build	2014-05-04 21:14:42.000000000 -0400
+++ linux-3.15-rc4/scripts/Makefile.build	2014-05-05 18:31:16.245756469 -0400
@@ -441,7 +441,7 @@
 cmd_link_multi-y = $(LD) $(ld_flags) -r -o $@ $(link_multi_deps) $(cmd_secanalysis)
 
 quiet_cmd_link_multi-m = LD [M]  $@
-cmd_link_multi-m = $(cmd_link_multi-y)
+cmd_link_multi-m = $(cmd_link_multi-y) /opt/intel/composer_xe_2013_sp1.2.144/compiler/lib/ia32/libirc_s.a
 
 # We would rather have a list of rules like
 # 	foo.o: $(foo-objs)
diff -uNr A/linux-3.15-rc4/scripts/Makefile.modpost linux-3.15-rc4/scripts/Makefile.modpost
--- A/linux-3.15-rc4/scripts/Makefile.modpost	2014-05-04 21:14:42.000000000 -0400
+++ linux-3.15-rc4/scripts/Makefile.modpost	2014-05-05 18:31:16.245756469 -0400
@@ -118,7 +118,7 @@
 quiet_cmd_ld_ko_o = LD [M]  $@
       cmd_ld_ko_o = $(LD) -r $(LDFLAGS)                                 \
                              $(KBUILD_LDFLAGS_MODULE) $(LDFLAGS_MODULE) \
-                             -o $@ $(filter-out FORCE,$^)
+                             -o $@ $(filter-out FORCE,$^) /opt/intel/composer_xe_2013_sp1.2.144/compiler/lib/ia32/libirc_s.a
 
 $(modules): %.ko :%.o %.mod.o FORCE
 	$(call if_changed,ld_ko_o)
